includes/utils.php

```diff
@@ -231,6 +231,54 @@ public static function cache_delete_group( $group ) {
 		}
 	}
 
+	/**
+	 * Handle transient WP options cache.
+	 *
+	 * @param array  $args
+	 * @param string $args[cache_key] The unique name for this cache
+	 * @param string $args[group] The name of the cache group
+	 *
+	 * @since x.x
+	 *
+	 * @return mixed
+	 */
+	public static function check_transient_cache( $args ) {
+		$results = get_transient( $args['group'] );
+		$key     = $args['cache_key'];
+		if ( false !== $results && is_array( $results ) && isset( $results[ $key ] ) ) {
```

Comment:
Let's simplify this down a bit:
if ( is_array( $results ) && isset( $results[ $key ] ) ) {